### PR TITLE
[fix][python] schedule unexpect offline each time update from api

### PR DESCRIPTION
**Pull request by zhongjiajie**
## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/python/PythonGateway.java [hunk 1]
```diff
@@ -348,6 +348,8 @@ private void createOrUpdateSchedule(User user,
             schedulerService.updateSchedule(user, projectCode, scheduleId, schedule, WarningType.valueOf(warningType),
                     warningGroupId, DEFAULT_FAILURE_STRATEGY, DEFAULT_PRIORITY, workerGroup, DEFAULT_ENVIRONMENT_CODE);
         }
+        // Always set workflow online to set schedule online
+        processDefinitionService.releaseProcessDefinition(user, projectCode, workflowCode, ReleaseState.ONLINE);
```
Comment (zhongjiajie) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/python/PythonGateway.java:352]: the key change of this PR is always set online of workflow

**Review by EricGao888 — APPROVED** [review]

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/SchedulerService.java [hunk 2]
```diff
@@ -128,12 +128,11 @@ Schedule getSchedule(User loginUser,
      * @param projectCode project code
      * @param id scheduler id
      * @param scheduleStatus schedule status
-     * @return publish result code
      */
-    Map<String, Object> setScheduleState(User loginUser,
-                                         long projectCode,
-                                         Integer id,
-                                         ReleaseState scheduleStatus);
+    void setScheduleState(User loginUser,
```
Comment (EricGao888) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/SchedulerService.java:132]: Just curious, why to make setScheduleState no more return value?

Comment (zhongjiajie) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/SchedulerService.java:132]: In the previous, we just return the error message inside the hashmap, we should avoid wrapper error but directly throw the exceptions in the code.

**Review by jieguangzhou — APPROVED** [review]

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/SchedulerService.java (hunk 2 above)
Comment (zhongjiajie) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/SchedulerService.java:132]: Using exception wrapper in hashmap will make the code too long and hard to test, you can see the previous `testSetScheduleState` in `SchedulerServiceTest.java` never test actual success

**Pull request merged**